src/app/blog/blob-store-design-challenges/page.mdx

```diff
@@ -0,0 +1,215 @@
+import { BlogPostLayout } from '@/components/BlogPostLayout'
+import {ThemeImage} from '@/components/ThemeImage'
+
+export const post = {
+  draft: false,
+  author: 'Rüdiger Klaehn',
+  date: '2025-02-24',
+  title: 'Blob store design challenges',
+  description:
+    "It's just hashed data, what's so complex about a blob store anyway?",
+}
+
+export const metadata = {
+  title: post.title,
+  description: post.description,
+}
+
+export default (props) => <BlogPostLayout article={post} {...props} />
+
+We've been hard at work with one of our protocols that you can pick off the shelf with iroh: [iroh-blobs]
+You might've used iroh-blobs before if you tried out [`sendme`](https://www.iroh.computer/sendme), which is just a small CLI wrapper over iroh with iroh-blobs.
+
+Put briefly, iroh-blobs helps transfer arbitrary data ("blobs") from one device to another via iroh in a way that allows these transfers to be interrupted, resumed and verified by content-addressing blobs using the BLAKE3 hashing function.
+
+Iroh-blobs also allows you to exchange the backing storage it uses via *blob store* abstraction, so the actual hashes and blobs might be stored as files on disk, stored fully in-memory or even fully remote by utilizing some cloud storage.
+
+Now, BLAKE3 hashed data and BLAKE3 verified streaming/bao are fully deterministic.
+
+The iroh blobs protocol is implemented in this spirit. Even for complex requests like requests that involve multiple ranges or entire hash sequences, for every request, there is exactly one sequence of bytes that is the correct answer. And the requester will notice if *data* is incorrect after at most 16 KiB of data.
```

Comment:
```suggestion
The iroh blobs protocol is implemented in this spirit. Even for complex requests that involve multiple ranges or entire hash sequences, there is exactly one sequence of bytes that is the correct answer. And the requester will notice if *data* is incorrect after at most 16 KiB of data.
```

I *think* this doesn't change the meaning of the sentence, so is still correct. But another read doesn't hurt.

Reply:
I want to emphasize that the fn from request to correct answer is fully deterministic (modulo limited ability to lie about sizes under certain circumstances), so I think keeping the `for every request`.